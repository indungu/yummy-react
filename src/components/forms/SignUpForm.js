import React, { Component } from 'react';
import { Redirect } from 'react-router-dom';
import { toast, ToastContainer } from 'react-toastify';
import axiosInstance from '../common/AxiosIntance';

export default class SignUpForm extends Component {
  constructor(props) {
    super(props);
    this.state = {
      username: '',
      email: '',
      password: '',
      confirmPassword: '',
      redirect: false,
    };

    this.onChange = this.onChange.bind(this);
    this.onSubmit = this.onSubmit.bind(this);
  }

  onChange(e) {
    this.setState({ [e.target.name]: e.target.value });
  }

  onSubmit(e) {
    e.preventDefault();
    if (this.state.password === this.state.confirmPassword) {
      // Get request data from form state.
      const requestData = {
        username: this.state.username,
        email: this.state.email,
        password: this.state.password,
      };

      // Make call to api through axios
      axiosInstance
        .post('/auth/register', requestData)
        .then((response) => {
          toast.success(response.data.message);
          this.setState({ redirect: true });
        })
        .catch((error) => {
          if (error.response) {
            toast.error(error.response.data.message);
          }
        });
    } else {
      toast.error('Password mismatch. Please try again');
    }
  }

  render() {
    if (this.state.redirect === true) {
      return <Redirect to="/login" />;
    }
    return (
      <div>
        <ToastContainer />
        <form className="auth-form" onSubmit={this.onSubmit}>
          <div className="form-group">
            <label className="control-label">Username</label>
            <input
              value={this.state.username}
              type="text"
              name="username"
              className="form-control"
              placeholder="John"
              onChange={this.onChange}
              autoComplete="current_username"
            />
          </div>
          <div className="form-group">
            <label className="control-label">Email</label>
            <input
              value={this.state.email}
              type="email"
              name="email"
              className="form-control"
              onChange={this.onChange}
              autoComplete="current_email"
              placeholder="john.doe@user.com"
            />
          </div>
          <div className="form-group">
            <label className="control-label">Password</label>
            <input
              value={this.state.password}
              type="password"
              name="password"
              className="form-control"
              onChange={this.onChange}
              autoComplete="current_password"
              placeholder="Password"
            />
          </div>
          <div className="form-group">
            <label className="control-label">Confirm Password</label>
            <input
              value={this.state.confirmPassword}
              type="password"
              name="confirmPassword"
              className="form-control"
              onChange={this.onChange}
              autoComplete="current_password"
              placeholder="Confirm Password"
            />
          </div>
          <span classNam="text-muted" style={{ fontSize: '9px', paddingBottom: '10% !important' }}>
            Password should contain at least one special character,
            One digit and one uppercase and lowercase letter.
            Password should not be less than 8 characters long
          </span>
          <div className="form-group">
            <button className="btn btn-primary btn-block">
              Sign Up
            </button>
          </div>
          <div className="well"><span>Already have an account? <a href="/login">Login</a></span></div>
        </form>
      </div>
    );
  }
}
