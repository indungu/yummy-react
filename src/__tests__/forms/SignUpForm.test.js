import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import SignUpForm from '../../components/forms/SignUpForm';

describe('SignUp form', () => {
  const wrapper = shallow(<SignUpForm />);
  // const preventDefault = jest.fn();
  it('Renders properly without crashing', () => {
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });

  // it('Renders with the correct initial state', () => {
  //   expect(wrapper.state().username.toEqual(''));
  //   expect(wrapper.state().email.toEqual(''));
  //   expect(wrapper.state().password.toEqual(''));
  //   expect(wrapper.state().confirmPassword.toEqual(''));
  // });

  // it('Can submit form data', () => {
  //   expect(wrapper.find('button#submitInfo')).to.have.length(1);
  //   expect(wrapper.instance().handleAddCategories({ preventDefault }));
  // });
});
