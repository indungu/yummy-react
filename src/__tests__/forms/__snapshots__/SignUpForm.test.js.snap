// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUp form Renders properly without crashing 1`] = `
<div>
  <ToastContainer
    autoClose={5000}
    bodyClassName={null}
    className={null}
    closeButton={
      <DefaultCloseButton
        ariaLabel="close"
      />
    }
    closeOnClick={true}
    hideProgressBar={false}
    newestOnTop={false}
    pauseOnHover={true}
    position="top-right"
    progressClassName={null}
    rtl={false}
    style={null}
    toastClassName={null}
    transition={[Function]}
  />
  <form
    className="auth-form"
    onSubmit={[Function]}
  >
    <div
      className="form-group"
    >
      <label
        className="control-label"
      >
        Username
      </label>
      <input
        autoComplete="current_username"
        className="form-control"
        name="username"
        onChange={[Function]}
        placeholder="John"
        required={true}
        type="text"
        value=""
      />
    </div>
    <div
      className="form-group"
    >
      <label
        className="control-label"
      >
        Email
      </label>
      <input
        autoComplete="current_email"
        className="form-control"
        name="email"
        onChange={[Function]}
        placeholder="john.doe@user.com"
        required={true}
        type="email"
        value=""
      />
    </div>
    <div
      className="form-group"
    >
      <label
        className="control-label"
      >
        Password
      </label>
      <input
        autoComplete="current_password"
        className="form-control"
        name="password"
        onChange={[Function]}
        placeholder="Password"
        required={true}
        type="password"
        value=""
      />
    </div>
    <div
      className="form-group"
    >
      <label
        className="control-label"
      >
        Confirm Password
      </label>
      <input
        autoComplete="current_password"
        className="form-control"
        name="confirmPassword"
        onChange={[Function]}
        placeholder="Confirm Password"
        required={true}
        type="password"
        value=""
      />
    </div>
    <span
      className="text-muted"
      style={
        Object {
          "fontSize": "9px",
          "paddingBottom": "10% !important",
        }
      }
    >
      Password should contain at least one special character, One digit and one uppercase and lowercase letter. Password should not be less than 8 characters long
    </span>
    <div
      className="form-group"
    >
      <button
        className="btn btn-primary btn-block"
        id="submitInfo"
        onClick={[Function]}
      >
        Sign Up
      </button>
    </div>
    <div
      className="well"
    >
      <span>
        Already have an account? 
        <a
          href="/login"
        >
          Login
        </a>
      </span>
    </div>
  </form>
</div>
`;
